Don't include malloc.h when it doesn't exist. Do include sys/socket.h.
https://sourceforge.net/p/xine/xine-lib-1.2/ci/e3f5170b04a872b58f87027d11c4e34a2b80462e/
--- src/input/http_helper.c.orig
+++ src/input/http_helper.c
@@ -24,7 +24,9 @@
 #include "config.h"
 #endif
 
+#ifdef HAVE_MALLOC_H
 #include <malloc.h>
+#endif
 #include <string.h>
 #include <stdint.h>
 
--- src/input/input_http.c.orig
+++ src/input/input_http.c
@@ -29,6 +29,9 @@
 #include <string.h>
 #ifdef HAVE_NETDB_H
 #include <netdb.h>
+#endif
+#ifdef HAVE_SYS_SOCKET_H
+#include <sys/socket.h>
 #endif
 #include <ctype.h>
 #include <errno.h>
--- src/input/tls/tls_gnutls.c.orig
+++ src/input/tls/tls_gnutls.c
@@ -41,7 +41,9 @@
 #endif
 
 #ifndef XINE_GNUTLS_3
-#  include <malloc.h>
+#  ifdef HAVE_MALLOC_H
+#    include <malloc.h>
+#  endif
 #  include <string.h>
 #  include <sys/stat.h>
 #  include <dirent.h>

